### DISCUSSION TITLE
AsyncExitStackMiddleware middleware.starlette takes a lot of time #2188

### DISCUSSION THREAD

**Discussion by theobouwman** [fastapi/fastapi#9689] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
I cant find this middleware anywhere. On the Starlette discussion they say its from FastAPI.
```

### Description

I am monitoring my API with Sentry and noted that AsyncExitStackMiddleware takes a lot of time... Is this normal?

<img width="1179" alt="Screenshot 2023-06-16 at 17 34 13" src="https://github.com/tiangolo/fastapi/assets/16098190/a86d6991-3a91-4143-a6d8-17d7d3d5c1d2">

### Operating System

Linux

### FastAPI Version

0.90.1

### Python Version

python3.11

**Comment by jgould22** [comment, marked as answer]
The middle ware is inserted here https://github.com/tiangolo/fastapi/blob/master/fastapi/applications.py#L201 and defined here https://github.com/tiangolo/fastapi/blob/master/fastapi/middleware/asyncexitstack.py#L7 (not that hard to find)

I have not used sentry with Fastapi but  this middleware wraps every request so if your span is timing the __call__ then it is going to measure the time spent in your controller, plus this middleware. If that is the case I would expect it to closely mirror the time of the request. If it is not the case then I do not know.

**Reply by theobouwman** [reply]
Okay clear. Thats the case. Thanks.
